src/util/Timer.h

```diff
@@ -82,10 +82,53 @@ class BaseTimer : public Timer {
     ReceiverMethod m_method;
 };
 
+template <class Receiver, bool kIsRepeating>
+class ArgBaseTimer : public Timer {
+public:
+    typedef void (Receiver::*ReceiverMethod)(void *data);
+
+    ArgBaseTimer()
+        : Timer(kIsRepeating)
+	, m_data(nullptr)
+        , m_receiver(nullptr)
+        , m_method(nullptr)
+    {
+    }
+
+    ~ArgBaseTimer() override
+    {
+        if (isRunning())
+            stop();
+    }
+
+    void handleCallback() override
+    {
+        running(kIsRepeating);
+        (m_receiver->*m_method)(m_data);
+    }
+
+    void start(int delayInMilliSeconds, Receiver* receiver, ReceiverMethod method, void *data = nullptr, bool willDestroy = false)
+    {
+        m_receiver = receiver;
+        m_method = method;
+	m_data = data;
+        Timer::start(delayInMilliSeconds, willDestroy);
+    }
+
+private:
+    void *m_data;
+    Receiver* m_receiver;
+    ReceiverMethod m_method;
```

Comment:
... Maybe, instead of all of this, you should just use std::bind, etc, so you pass a callback and you define the extra argument or arguments from callback creation instead of here.

Reply:
Sure, imagine the signature of a callback that will be invoked is like this:
`typedef void (*Callback)(int a, bool b)`

And you want to pass a assign the callback passing the this pointer for a class implementation. The method you create could be:

```
class C {
 public:
  void MyCallbackImpl(int a, bool b);
}

Then, assigning the callback would be:
```
Callback callback = std::bind(&C::MyCallbackImpl, this, std::placeholders::_1, std::placeholders::_2);
```

The idea is that you store in the callback (with bind) all the data you need to store, so instead of adding receiver and data to the class, it is bound to the variable.

I am not 100% sure if this is exactly used like this (as in Chromium we have base::BindOnce and other helpers with a slightly different behavior). But that's the idea.